Tests/SwerverTests/ResponseSpec/HttpResponseSpec.swift

```diff
@@ -5,48 +5,104 @@ import Foundation
 
 class HttpResponseSpec: QuickSpec {
     override func spec() {
-        describe("HttpResponse") {
-            var responseBuilder = ResponseBuilder()
+        describe("A Reponse object") {
+            func summonBuilder() -> HttpResponseBuilder {
+                return HttpResponseBuilder()
+            }
             
-            it("can equate two response object") {
-                let responseOne = responseBuilder
-                    .with(statusCode: .ok)
-                    .with(contentType: .text)
-                    .with(body: String())
-                    .build()
-                
-                let responseTwo = responseBuilder
-                    .with(statusCode: .ok)
-                    .with(contentType: .text)
-                    .with(body: String())
-                    .build()
-                
-                expect(responseOne).to(equal(responseTwo))
+            context("when compares with another response object") {
+                it("is true when attributes match") {
+                    var responseBuilder = summonBuilder()
+                    
+                    let responseOne = TestData.buildResponse()
+                    let responseTwo = responseBuilder
+                        .with(statusCode: .ok)
+                        .with(contentType: .text)
+                        .set(header: .allow, value: "GET")
+                        .with(body: "Hello")
+                        .build()
+                    
+                    expect(responseOne).to(equal(responseTwo))
+                }
+                
+                it("is false when the status code does not match") {
+                    var responseBuilder = summonBuilder()
+                    
+                    let responseOne = TestData.buildResponse()
```

Comment:
This test has two issues:
1. Details we care about are hidden from us
   * The differing status code
2. Details we do not directly care about are making the test noisy.
   * All the extra attributes

In other words, right now, you say the `"status code does not match"`, but it is hard to see that in the test.

## Examples

Here are two different examples of possible refactorings. Both of these changes make it so the test exposes the details you care about (the differing status code), and abstracts away or removes the details you do not.

### Removing Extraneous Details

This example removes the details we not care about. If some of the attributes are required, this does not work.

```swift
it("is false when the status code does not match") {
    var responseBuilder = summonBuilder()
    let responseOne = responseBuilder
      .with(statusCode: .not_found)
      .build()
    let responseTwo = responseBuilder
      .with(statusCode: .ok)
      .build()

    expect(responseOne).toNot(equal(responseTwo))
}
```

### Abstracting Extraneous Details

If you need or want to provide default values for some of the attributes, you can leverage the builder pattern here.

```swift
func defaultResponseBuilder() -> HttpResponseBuilder {
    return HttpResponseBuilder()
        .with(statusCode: .not_found)
        .with(contentType: .text)
        .set(header: .allow, value: "GET")
        .with(body: "Hello")
        .build()
}

it("is false when the status code does not match") {
    let responseOne = defaultResponseBuilder()
      .with(statusCode: .not_found)
      .build()
    let responseTwo = defaultResponseBuilder()
      .with(statusCode: .ok)
      .build()

    expect(responseOne).toNot(equal(responseTwo))
}
```

Reply:
Oops! `build` should not be called in `defaultResponseBuilder`.

Reply:
will fix that